sdk/src/sharedTest/java/io/customer/sdk/CustomerIOClientTest.kt

```diff
@@ -253,6 +253,44 @@ class CustomerIOClientTest : BaseTest() {
         prefRepository.getDeviceToken() shouldBeEqualTo givenDeviceToken
     }
 
+    // addCustomDeviceAttributes
+
+    @Test
+    fun addCustomDeviceAttributes_givenNoPushToken_expectDoNotRegisterPushToken() {
+        val givenAttributes = mapOf(String.random to String.random)
+
+        customerIOClient.addCustomDeviceAttributes(givenAttributes)
+
+        // no token registered
+        verifyNoInteractions(backgroundQueueMock)
```

Comment:
I do not like the code used here to test if `registerDeviceToken()` was called. This will be addressed with [a refactor splitting up the `CustomerIOClient` class](https://github.com/customerio/issues/issues/7160). These lines of code could then be much easier to understand and easier to maintain. I imagine the code will be changed to something similar to this:

```kotlin
verify(deviceRepository, never()).registerDeviceToken(givenDeviceToken)
```

Reply:
implemented https://github.com/customerio/customerio-android/pull/103